cli.py

```diff
@@ -134,18 +134,24 @@ def load_bouquets(env: str = "demo", include_private: bool = False):
 def process_dataset(env: str, d: dict, licenses: list, skip_related: bool) -> None:
     """Process a single dataset and its resources"""
     prefix = get_config_value(env, "prefix")
-    if organization_id := (d.get("organization") or {}).get("id"):
-        load_organization(env, organization_id)
-
-    dataset_obj = Dataset.from_payload(d, prefix, licenses)
-    existing = app.session.query(Dataset).filter_by(dataset_id=dataset_obj.dataset_id).first()
-    upsert(app.session, dataset_obj, existing)
-
-    if not skip_related:
-        for r in iter_rel(d["resources"], quiet=True):
-            resource_obj = Resource.from_payload(r, dataset_obj.dataset_id)
-            app.session.add(resource_obj)
-        app.session.commit()
+    try:
+        if organization_id := (d.get("organization") or {}).get("id"):
+            load_organization(env, organization_id)
+
+        dataset_obj = Dataset.from_payload(d, prefix, licenses)
+        existing = app.session.query(Dataset).filter_by(dataset_id=dataset_obj.dataset_id).first()
+        upsert(app.session, dataset_obj, existing)
+
+        if not skip_related:
+            for r in iter_rel(d["resources"], quiet=True):
+                resource_obj = Resource.from_payload(r, dataset_obj.dataset_id)
+                app.session.add(resource_obj)
+            app.session.commit()
+    except Exception as e:
+        app.session.rollback()
+        if sentry_dsn:
```

Comment:
Is there a way to connect to sentry without such a tight dependency? Like a global init that would log all uncaught exception to sentry?

Reply:
It's what sentry does by default. But we're catching the Exception in the thread pool manager, and sentry error management for a thread outside of the thread is flaky at best.